### PR TITLE
Features: flow, invert, mapValues, uniqBy

### PR DESCRIPTION
**Pull request by dictor93**
### Description:
Added features:
- `flow` which generates a composite function that returns the result of provided functions called by the chain; each previous function result passes as the argument of the next function in the chain
- `invert` - Inverts the key-value pairs of an object
- `mapValues` - maps the values of an object or array using the provided iteratee function or property path
- `uniqBy` - get unique values of array by the iteratee function or property path

### New features:
- [x] flow
- [x] invert
- [x] mapValues
- [x] uniqBy

### PR status:
- [x] Version bumped
- [x] Change-log updated
- [x] Tests added or updated
- [x] PR keeps 100% test coverage
- [x] Type definition updated
- [x] Readme updated
- [x] Linter is applied
- [x] Const arrow functions are used instead of pure function definitions where applicable
- [x] Functions are listed in alphabetic order in index.js, index.d.ts and readme

[Related PR](https://github.com/bitfinexcom/bfx-core-kyc-js/pull/1185)

### CODE REVIEW

File: src/flow.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+/**
+ * Executes a series of functions in a pipeline, passing the result of each function as the argument to the next function.
+ *
+ * @param {Array<Function>} fns - An array of functions to be executed in order.
+ * @returns {any} - The final result of the pipeline.
+ * @throws {Error} - Throws an error if any element in the array is not a function.
+ */
+const flow = (fns = []) => {
+  if (fns.find(fn => typeof fn !== 'function')) throw new Error('All elements should be functions')
+  return (arg) => fns.reduce((acc, fn) => fn(acc), arg)
```
Comment (grantfayvor) [inline src/flow.js]: lodash version of this function accepts an arbitrary number of arguments. so maybe something like
```suggestion
  return (...args) => fns.reduce((acc, fn) => [fn(...acc)], args)
```

**Review by grantfayvor — COMMENTED** [review]
small comment

**Review by grantfayvor — APPROVED** [review]
lgtm!

File: src/invert.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+'use strict'
+
+/**
+ * Inverts the key-value pairs of an object.
+ *
+ * @param {object} object - The object to invert.
+ * @returns {object} - The new object with inverted key-value pairs.
+ */
+const invert = (object) =>
+  Object.fromEntries(Object.entries(object).map(([key, value]) => [value, key]))
```
Comment (vigan-abd) [inline src/invert.js]: This approach has more lines of code but is more efficient as it has just a single loop
```suggestion
const invert = (object) => {
  const newObject = {}
  for (const key in object) {
    newObject[object[key]] = key
  }
  return newObject
}
```

**Review by vigan-abd — CHANGES_REQUESTED** [review]

**Review by vigan-abd — APPROVED** [review]

**Review by vigan-abd — APPROVED** [review]

**Review by rjmacarthy — APPROVED** [review]

**Review by ezewer — APPROVED** [review]
looks good to me!

**Pull request merged**